apps/blockchain/lib/blockchain/account.ex

```diff
@@ -545,24 +545,55 @@ defmodule Blockchain.Account do
       iex> Blockchain.Account.get_storage(updated_state, <<01::160>>, 5)
       {:ok, 9}
   """
-  @spec put_storage(EVM.state(), Address.t(), integer(), integer()) :: EVM.state()
-  def put_storage(state, address, key, value) do
-    update_account(state, address, fn acct ->
-      {updated_storage_trie, updated_trie} = Storage.put(state, acct.storage_root, key, value)
-      root_hash = TrieStorage.root_hash(updated_storage_trie)
+  @spec put_storage(
+          TrieStorage.t(),
+          Address.t() | {Address.t(), t()},
+          integer(),
+          integer(),
+          boolean()
+        ) :: {t(), TrieStorage.t()} | t()
+  def put_storage(state, address, key, value, return_account \\ false)
```

Comment:
Since `state` is now `TrieStorage.t()`, should the parameter be updated to `trie` in this and the below functions?